SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialShadowActor.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma optimize("", off)
+#include "EngineClasses/SpatialShadowActor.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialShadowActor);
+
+USpatialShadowActor::USpatialShadowActor(const FObjectInitializer& ObjectInitializer /*= FObjectInitializer::Get()*/)
+	: Super(ObjectInitializer)
+	, EntityId(SpatialConstants::INVALID_ENTITY_ID)
+	, ReplicatedPropertyHash("")
+
+{
+}
+
+void USpatialShadowActor::Init(const Worker_EntityId InEntityId, AActor* InActor)
+{
+	check(InEntityId != SpatialConstants::INVALID_ENTITY_ID);
+	check(InActor != nullptr);
+	check(IsValid(InActor));
+
+	EntityId = InEntityId;
+	ReplicatedPropertyHash = CreateHash(InActor);
+	Actor = InActor;
+}
+
+void USpatialShadowActor::Update(const Worker_EntityId InEntityId, AActor* InActor)
+{
+	check(InActor == Actor);
+
+	check(InEntityId != SpatialConstants::INVALID_ENTITY_ID);
+	check(InEntityId == EntityId);
+	check(InActor != nullptr);
+
+	ReplicatedPropertyHash = CreateHash(InActor);
+	Actor = InActor;
+}
+
+FString USpatialShadowActor::CreateHash(AActor* InActor)
```

Comment:
function should be const, and parameter should be a ref (and hopefully a const ref, if code below allows)?

Reply:
After refactor this function can no longer be const and already changed to const actor ref.